src/SourceLocation.ts

```diff
@@ -0,0 +1,51 @@
+// LICENSE : MIT
+"use strict";
+import SourceCode from "structured-source";
+import { RuleErrorPadding } from "./wrapIgnore";
+import { TxtNode, TxtParentNode } from "@textlint/ast-node-types";
+
+export class SourceLocation {
```

Comment:
This should be used with textlint's core logic.
Currently, textlint does not export this logic.